scripts/images-exporter/images-exporter.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+import docker
+import argparse
+import yaml
+import logging
+import os
+import shutil
+
+logging.basicConfig(format='%(asctime)s %(message)s', level=logging.INFO)
+
+class Image(object):
+    def __init__(self, image = '', repository = None, tag = None) -> None:
+        super().__init__()
+        if repository != None and tag != None:
+            self.repository = repository
+            self.tag = tag
+            return
+        parts = image.split(':')
+        if len(parts) != 2:
+            raise ValueError(f"invalid image {image}")
+        self.repository = parts[0]
+        self.tag = parts[1]
+    
+    def get_repo_name(self) -> str:
+        pos = self.repository.rfind('/')
+        if pos < 0:
+            return self.repository
+        else:
+            return self.repository[pos + 1:]
+    
+    def __str__(self) -> str:
+        return f"{self.repository}:{self.tag}"
+
+
+class RegistryImageSink(object):
+    def __init__(self, docker_client, target_registry) -> None:
+        super().__init__()
+        self.docker_client = docker_client
+        self.target_registry = target_registry
```

Comment:
In some cases, users just want to pull the required images but no need to change the registry. So, i think target_registry should be a optiontal config.